x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformPersistentTasksExecutor.java

```diff
@@ -100,7 +101,10 @@ public TransformPersistentTasksExecutor(Client client,
             return new PersistentTasksCustomMetaData.Assignment(null, reason);
         }
         DiscoveryNode discoveryNode = selectLeastLoadedNode(clusterState, (node) ->
-            node.isDataNode() && node.getVersion().onOrAfter(params.getVersion())
+            node.isDataNode() &&
+            // see gh#48019 older nodes might not be able to read documents
+            node.getVersion().onOrAfter(Version.V_7_4_0) &&
+            node.getVersion().onOrAfter(params.getVersion())
```

Comment:
I think we should not allow assignment at all in a mixed cluster where a `<= 7.3` node is present. If the task is running and somebody makes a stats call, that stats call could be made against a 7.3 node which queries the wrong index and incorrectly says that the transform does not exist.